### PR TITLE
[KYUUBI #2360] [Subtask] Configuring Hive engine heap memory and java opts

### PR DESCRIPTION
**Pull request by zhaomin1423**
### _Why are the changes needed?_

Configuring Hive engine heap memory and java opts

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
it conflicts

**Comment by zhaomin1423** [comment]
> it conflicts

done

**Comment by yaooqinn** [comment]
thanks, merged to master

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/hive/HiveProcessBuilder.scala [hunk 1]
```diff
@@ -75,20 +81,21 @@ class HiveProcessBuilder(
     env.get("YARN_CONF_DIR").foreach(classpathEntries.add)
     // jars from hive distribution
     classpathEntries.add(s"$hiveHome${File.separator}lib${File.separator}*")
-    val hadoopCp = env.get("HIVE_HADOOP_CLASSPATH").orElse(env.get("HADOOP_CLASSPATH"))
-    hadoopCp.foreach(path => classpathEntries.add(s"$path${File.separator}*"))
-    if (hadoopCp.isEmpty) {
-      warn(s"HIVE_HADOOP_CLASSPATH or HADOOP_CLASSPATH don't export.")
+    val extraCp = conf.get(ENGINE_HIVE_EXTRA_CLASSPATH)
```
Comment (deadwind4) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/hive/HiveProcessBuilder.scala:84]: Now, HiveSQLEngine is coupled of `kyuubi.engine.hive.extra.classpath` in production environment. If we don't set `kyuubi.engine.hive.extra.classpath` the HiveSQLEngine will error.

Comment (deadwind4) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/hive/HiveProcessBuilder.scala:84]: Now, HiveSQLEngine is coupled of `kyuubi.engine.hive.extra.classpath` in production environment. If we don't set `kyuubi.engine.hive.extra.classpath` the HiveSQLEngine will error.

Comment (zhaomin1423) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/hive/HiveProcessBuilder.scala:84]: it is an expectant behaviour.

Comment (deadwind4) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/hive/HiveProcessBuilder.scala:84]: The` kyuubi.engine.hive.extra.classpath` is required in production environment. I'm not sure it's user-friendly. We can discuss in this issue. https://github.com/apache/incubator-kyuubi/issues/2433

**Comment by zhaomin1423** [comment]
hadoop classpath has some unnecessary dependicies for hive engine, it may be lead to some class conflicts in the future.



| |
***@***.***
|
|
***@***.***
|




---- 回复的原邮件 ----
| 发件人 | Ada ***@***.***> |
| 日期 | 2022年04月21日 16:13 |
| 收件人 | ***@***.***> |
| 抄送至 | Xiao ***@***.******@***.***> |
| 主题 | Re: [apache/incubator-kyuubi] [KYUUBI #2360] [Subtask] Configuring Hive engine heap memory and java opts (PR #2371) |

@deadwind4 commented on this pull request.

In kyuubi-server/src/main/scala/org/apache/kyuubi/engine/hive/HiveProcessBuilder.scala:

> @@ -75,20 +81,21 @@ class HiveProcessBuilder(
     env.get("YARN_CONF_DIR").foreach(classpathEntries.add)
     // jars from hive distribution
     classpathEntries.add(s"$hiveHome${File.separator}lib${File.separator}*")
-    val hadoopCp = env.get("HIVE_HADOOP_CLASSPATH").orElse(env.get("HADOOP_CLASSPATH"))
-    hadoopCp.foreach(path => classpathEntries.add(s"$path${File.separator}*"))
-    if (hadoopCp.isEmpty) {
-      warn(s"HIVE_HADOOP_CLASSPATH or HADOOP_CLASSPATH don't export.")
+    val extraCp = conf.get(ENGINE_HIVE_EXTRA_CLASSPATH)


The kyuubi.engine.hive.extra.classpath is required in production environment. I'm not sure it's user-friendly. We can discuss in this issue. #2433

**Pull request closed without merging**